search-app/src/cmr/search/services/autocomplete_service.clj

```diff
@@ -0,0 +1,31 @@
+(ns cmr.search.services.autocomplete-service
+  "Service for autocomplete functionality"
+  (:require
+    [clojure.string :as str]
+    [cheshire.core :as json]
+    [cmr.common-app.services.search.params :as common-params]
+    [cmr.common-app.services.search.query-execution :as qe]
+    [cmr.common-app.services.search.query-model :as qm]
+    [cmr.common-app.services.search.group-query-conditions :as gc]))
+
+(defn autocomplete
+  "Execute elasticsearch query to get autocomplete suggestions"
+  ([context term]
+   (autocomplete context term nil))
+  ([context term opts]
+   (let [condition (if (empty? (:types opts))
```

Comment:
A bit of a nitpick: it's best to avoid `if` statements inside of `let`s. You can probably just split this `if` into a function of its own pretty easily
```suggestion
   (let [condition (build-autocomplete-condition term opts)
```